Configurations/README

```diff
@@ -101,6 +101,25 @@ In each table entry, the following keys are significant:
                            files.  On unix, this defaults to "" (NOTE:
                            this is here for future use, it's not
                            implemented yet)
+        shlib_variant   => A "variant" identifier inserted between the base
+                           shared library name and the extension.  This
+                           supports installation of custom OpenSSL libraries
+                           that don't conflict with OpenSSL libraries provided
+                           by the operating system.  The variant identifier
+                           becomes part of the SONAME of the library and also
+                           any symbol versions.  For example, on a system where
+                           a default build would normally create the SSL shared
+                           library as 'libssl.so -> libssl.so.1.1' with the
+                           value of the symlink as the SONAME, a target
+                           definition that sets 'shlib_variant => "-abc"' will
+                           create 'libssl.so -> libssl-abc.so.1.1', again with
+                           an SONAME equal to the value of the symlink.  The
+                           symbol versions associated with the variant library
+                           would then be 'OPENSSL_ABC_<version>' rather than
+                           the default 'OPENSSL_<version>'. The string inserted
+                           into symbol versions is obtained by mapping all
+                           letters in the "variant" identifier to upper case
+                           and all non-alphanumeric characters to '_'.
```

Comment:
Same note as in the 1.1.0 PR:

It should be noted that this is available on select platforms only.  We only have symbol versions on Linux and Solaris so far, and the SONAME hack is Unix / MacOS X only.  Windows and VMS be damned in this case.